o2a/o2a_libs/el_parser.py

```diff
@@ -0,0 +1,296 @@
+# -*- coding: utf-8 -*-
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This module contains tools for translating Expression Language to Jinja.
+
+The BNF for the grammar is based on official grammar of EL:
+https://download.oracle.com/otn-pub/jcp/jsp-2.1-fr-spec-oth-JSpec/jsp-2_1-fr-spec-el.pdf
+"""
+
+__all__ = ["translate"]
+
+from typing import Union
+import re
+
+from lark import Lark, Tree, Token
+
+
+GRAMMAR = r"""
```

Comment:
Is this just a reproduction of the JSP lang spec? e.g., 

LValue ::= ‘${‘ LValueInner ‘}’
| ‘#{‘ LValueInner ‘}’
...

Reply:
Exactly, those production are based on the specification. However, the grammar is changed a little bit to simplify it and avoid some recursion errors.